sklearn/datasets/_samples_generator.py

```diff
@@ -1149,7 +1163,8 @@ def make_low_rank_matrix(n_samples=100, n_features=100, effective_rank=10,
     return np.dot(np.dot(u, s), v.T)
 
 
-def make_sparse_coded_signal(n_samples, n_components, n_features,
+@_deprecate_positional_args
+def make_sparse_coded_signal(n_samples, n_components, *, n_features,
```

Comment:
Because the other `make*` has `n_features` as the second argument:

```suggestion
def make_sparse_coded_signal(n_samples, *, n_components, n_features,
```